nixos/tests/gitea.nix

```diff
@@ -45,7 +45,7 @@ with pkgs.lib;
       {
         services.gitea.enable = true;
         services.gitea.database.type = "postgres";
-        services.gitea.database.password = "secret";
+        services.gitea.database.passwordFile = pkgs.writeText "db-password" "secret";
```

Comment:
Not to get into a discussion about the value of the warning itself, but using `writeText` here puts the secret into the Nix store.  While it may do the job of silencing the warning (incorrectly so, IMO) there's a risk that somebody might take this as an example of how to use the service.  I believe the "correct" solution is to point to a file wholly outside the purview of Nix (e.g., `"/var/lib/gitea/secret.key"`).

Reply:
I understand the educational concern, but this is done ~everywhere in tests already. Grepping for pkgs.writeText:

```
ldap.nix:      environment.etc."nslcd.rootpwmodpw".source = pkgs.writeText "rootpwmodpw" dbAdminPwd;
ldap.nix:      environment.etc."ldap/bind.password".source = pkgs.writeText "password" dbAdminPwd;
ssh-keys.nix:{ snakeOilPrivateKey = pkgs.writeText "privkey.snakeoil" ''
borgbackup.nix:  privateKey = pkgs.writeText "id_ed25519" ''
borgbackup.nix:  privateKeyAppendOnly = pkgs.writeText "id_ed25519" ''
nextcloud/with-postgresql-and-redis.nix:          dbpassFile = toString (pkgs.writeText "db-pass-file" ''
nextcloud/with-postgresql-and-redis.nix:          adminpassFile = toString (pkgs.writeText "admin-pass-file" ''
gitolite.nix:  adminPrivateKey = pkgs.writeText "id_ed25519" ''
gitolite.nix:  alicePrivateKey = pkgs.writeText "id_ed25519" ''
gitolite.nix:  bobPrivateKey = pkgs.writeText "id_ed25519" ''
```

Would you be fine with just adding a comment that says "don't do that in production"?

Reply:
A warning note would be fine, but if we don't care about the underlying issue I suppose we can just merge this as-is (I will note however that if these modules start enforcing the no-secrets-in-store rule properly then I'd expect most of the above to become errors/warnings). 

I generally don't agree that an existing bad practice is reason to add more of the same, but I'm happy to defer to your judgement.